s2spy/preprocess.py

```diff
@@ -88,12 +123,56 @@ def _check_input_data(data: Union[xr.DataArray, xr.Dataset]):
         )
 
 
+def _check_temporal_resoltuion(timescale: TIMESCALE_TYPE) -> TIMESCALE_TYPE:
+    support_temporal_resolution = ["monthly", "weekly", "daily"]
+    if timescale not in support_temporal_resolution:
+        raise ValueError(
+            "Given temporal resoltuion is not supported."
+            "Please choose from 'monthly', 'weekly', 'daily'."
+        )
+    return timescale
+
+
+def _check_data_resolution_match(
+    data: Union[xr.DataArray, xr.Dataset], timescale: TIMESCALE_TYPE
+):
+    """Check if the temporal resoltuion of input is the same as given timescale."""
+    timescale_dict = {
+        "monthly": np.timedelta64(1, "M"),
+        "weekly": np.timedelta64(1, "W"),
+        "daily": np.timedelta64(1, "D"),
+    }
+    time_intervals = np.diff(data["time"].to_numpy())
+    temporal_resolution = np.median(time_intervals).astype("timedelta64[D]")
+    if timescale == "monthly":
+        temporal_resolution = temporal_resolution.astype(int)
+        if temporal_resolution <= 31 and temporal_resolution >= 28:
+            pass
+        else:
```

Comment:
```suggestion
        min_days, max_days = (28, 31)
        if min_days > temporal_resolution > max_days:
```
I think this is more clear and straightforward. I predefined min_days and max_days to make it a bit more explicit.

I am also not sure to what extent we should "babysit" the user. If they throw in a quarterly timeseries, I feel that's their own problem :sweat_smile:

Reply:
Yep, this looks much nicer! Thanks!

Reply:
Wait, just notice that `if min_days > temporal_resolution > max_days:` actually equals to ` min_days > temporal_resolution and temporal_resolution > max_days:`. It is not `or`. Should be the following:
```py
        min_days, max_days = (28, 31)
        if not max_days > temporal_resolution > min_days:
            warnings.warn(
                "The temporal resolution of data does not completely match "
                "the target timescale. Please check your input data.",
                stacklevel=1,
            )
```